crates/bevy_ecs/src/schedule/ambiguity_detection.rs

```diff
@@ -0,0 +1,812 @@
+use crate::component::ComponentId;
+use crate::schedule::{AmbiguityDetection, SystemContainer, SystemStage};
+use crate::world::World;
+
+use bevy_utils::get_short_name;
+use bevy_utils::tracing::{error, warn};
+use fixedbitset::FixedBitSet;
+use std::hash::Hash;
+
+/// Systems that access the same Component or Resource within the same stage
+/// risk an ambiguous order that could result in logic bugs, unless they have an
+/// explicit execution ordering constraint between them.
+///
+/// This occurs because, in the absence of explicit constraints, systems are executed in
+/// an unstable, arbitrary order within each stage that may vary between runs and frames.
```

Comment:
```suggestion
/// Within a stage, writes to a component or resource (and reads relative to writes) should typically have a well-defined order.
///
/// Ambiguous order between systems that have conflicting access can result in subtle logic bugs since,
/// in the absence of constraints, systems are executed in an arbitrary order and that can change between runs
/// and even frames.
```